test_spectral_norm.py

```diff
@@ -100,7 +100,7 @@ def tf_fwd(inp: tf.Tensor, label: tf.Tensor,
     if spectral_normalization:
         conv = network_components.SpectralNormalization(
                 tf.keras.layers.Conv2D(1, kernel_size=(3, 3), strides=(1, 1),
-                weights=[W_tf], trainable=False, padding='same',
+                weights=[W_tf], trainable=True, padding='same',
```

Comment:
turns out either way the tests pass, oddly. fwiw the default is true, and in my feature branch i've just refactored all of this so it won't matter.

Reply:
it can just be trainable, we won't use the weights in the forward pass so the module will never recognize them as variables